src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,42 @@
+const { ApiError } = require('../exeptions/auth.error');
+const { jwtServices } = require('../services/jwt.services');
+const { usersServices } = require('../services/users.services');
+
+module.exports = {
+  authMiddleware: async (req, _, next) => {
+    const authorizationHeader = 'authorization';
+
+    const token =
+      (req.headers[authorizationHeader] &&
+        req.headers[authorizationHeader].split(' ')[1]) ||
+      req.cookies.token;
+
+    if (!token) {
+      throw ApiError.unauthorized('Authorization token is missing');
+    }
+
+    const refreshKey = process.env.JW_REFRESH_KEY;
+
+    if (!refreshKey) {
+      throw new Error('Environment variable JW_REFRESH_KEY is not set');
+    }
+
+    let email;
+
+    try {
+      ({ email } = jwtServices.verify(token, refreshKey));
```

Comment:
The middleware uses `JW_REFRESH_KEY` for verifying the token. Ensure that this is the intended behavior, as typically, access tokens and refresh tokens use different keys. Consider using the appropriate key for the type of token being verified.